airbyte-webapp/src/components/ui/CollapsablePanel/CollapsablePanel.tsx

```diff
@@ -0,0 +1,49 @@
+import { faChevronRight, faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import React, { ReactElement } from "react";
+import { useToggle } from "react-use";
+
+import styles from "./CollapsablePanel.module.scss";
+
+const Arrow: React.FC<{ isOpen: boolean }> = ({ isOpen }) => (
+  <FontAwesomeIcon icon={isOpen ? faChevronDown : faChevronRight} />
+);
+
+/**
+ * CollapsablePanel defines a reusable "click to expand" UI element. It has two
+ * props for content slots, each containing an arbitrary component tree:
+ *   - children: the "above the fold" or "title" content which is always shown
+ *   - drawer: the "below the fold" content which is expanded or hidden on user interaction
+ *
+ * The rationale for putting the "above the fold" content in `children` is that
+   react's syntactic sugar for `children` makes the source code better reflect
+   the rendered UI, with the always-present content defined inline next to
+   its "peer" components in the surrounding context.
+ */
+export const CollapsablePanel: React.FC<{
+  drawer: ReactElement;
+  initiallyOpen?: boolean;
+  className?: string;
+  openClassName?: string;
+  closedClassName?: string;
+}> = ({ drawer, initiallyOpen = false, children, className, openClassName, closedClassName }) => {
+  const [isOpen, toggleOpen] = useToggle(initiallyOpen);
+  return (
+    <div
+      className={classNames({
+        [className || ""]: true,
+        [openClassName || ""]: isOpen,
+        [closedClassName || ""]: !isOpen,
+      })}
+    >
+      <div>
+        <button className={styles.arrow} tabIndex={0} onClick={toggleOpen}>
```

Comment:
I think we might simply want to use the https://headlessui.com/react/disclosure component from headless UI, which we use in other places for collapsible elements, which handles all the accessibility. I know that the platform workflow team has also thought about creating a component for that now. @edmundito @dizel852 do you know if this is already started work on, since we should try to coordinate do not do that effort duplicate.

Reply:
If we keep using `button` here it also won't require `tabIndex`, since that's the default value of the `button` element anyway.

Reply:
> UX idea: add <label>{children}</label> so you can click on "Advanced options" as well as just the arrow button.

Or children should just be part of the `button` completely. That one got me as well. Since I feel usually there is more content inside the collapsible than in the button, I'd suggest we make the `drawer` property just to be the `children` instead and add a `buttonLabel` property instead that renders inside the button, so we don't need to put the full content of the collapsiple into a property.